src/probnum/diffeq/odefiltsmooth/odefiltsmooth.py

```diff
@@ -218,196 +254,187 @@ def probsolve_ivp(
-            firststep = 0.01 * norm_y0 / norm_dy0
+        firststep = step if step is not None else firststep
         stprl = steprule.AdaptiveSteps(firststep=firststep, atol=atol, rtol=rtol)
     return stprl
 
 
-def _check_step_tol(step, atol, rtol):
-    both_none = atol is None and rtol is None and step is None
-    both_not_none = (atol is not None and rtol is not None) and step is not None
-    if both_none or both_not_none:
-        errormsg = "Please specify either a tolerance or a step size."
-        raise ValueError(errormsg)
-    atol_not_rtol = atol is not None and rtol is None
-    rtol_not_atol = rtol is not None and atol is None
-    if atol_not_rtol or rtol_not_atol:
-        errormsg = "Please specify either both atol and rtol, or neither."
-        raise ValueError(errormsg)
-
-
-def _string2prior(ivp, which_prior, **kwargs):
+def _string2prior(ivp, which_prior, driftspeed, lengthscale):
 
-    ibm_family = ["ibm1", "ibm2", "ibm3", "ibm4"]
-    ioup_family = ["ioup1", "ioup2", "ioup3", "ioup4"]
-    matern_family = ["matern32", "matern52", "matern72", "matern92"]
+    ibm_family = ["IBM1", "IBM2", "IBM3", "IBM4"]
+    ioup_family = ["IOUP1", "IOUP2", "IOUP3", "IOUP4"]
+    matern_family = ["MAT32", "MAT52", "MAT72", "MAT92"]
     if which_prior in ibm_family:
-        return _string2ibm(ivp, which_prior, **kwargs)
+        return _string2ibm(ivp, which_prior)
     elif which_prior in ioup_family:
-        return _string2ioup(ivp, which_prior, **kwargs)
+        return _string2ioup(ivp, which_prior, driftspeed=driftspeed)
     elif which_prior in matern_family:
-        return _string2matern(ivp, which_prior, **kwargs)
-    else:
-        raise RuntimeError("It should have been impossible to reach this point.")
+        return _string2matern(ivp, which_prior, lengthscale=lengthscale)
+    raise ValueError("This prior is not supported.")
 
 
-def _string2ibm(ivp, which_prior, **kwargs):
+def _string2ibm(ivp, which_prior):
 
-    if which_prior == "ibm1":
+    if which_prior == "IBM1":
         return pnfs.statespace.IBM(
             1,
             ivp.dimension,
             forward_implementation="sqrt",
             backward_implementation="sqrt",
         )
-    elif which_prior == "ibm2":
+    elif which_prior == "IBM2":
         return pnfs.statespace.IBM(
             2,
             ivp.dimension,
             forward_implementation="sqrt",
             backward_implementation="sqrt",
         )
-    elif which_prior == "ibm3":
+    elif which_prior == "IBM3":
         return pnfs.statespace.IBM(
             3,
             ivp.dimension,
             forward_implementation="sqrt",
             backward_implementation="sqrt",
         )
-    elif which_prior == "ibm4":
+    else:  # which_prior == "IBM4":
         return pnfs.statespace.IBM(
             4,
             ivp.dimension,
             forward_implementation="sqrt",
             backward_implementation="sqrt",
         )
```

Comment:
this could in principle be replaced with  `ordint = which_prior[-1]` which would remove all the ifs. This goes together with the comment above I guess. Thoughts?

Reply:
I would strongly prefer extracting the order from the string and then using that, to remove all the duplicate code here. We might need to take some special care to ensure, that people don't supply values >= 10, but if we keep the explicit string matching that we currently have in `_string2prior` then that's not a risk.